libs/types/src/cdf/election-results-reporting/convert.ts

```diff
@@ -0,0 +1,269 @@
+  for (const selection of contest.ContestSelection as ResultsReporting.CandidateSelection[]) {
+    const candidateId = selection['@id'];
+    tallies[candidateId] = {
+      id: candidateId,
+      name: candidateNameRecord[candidateId],
+      tally: findTotalVoteCounts(assertDefined(selection.VoteCounts)),
+      // partyIds: unimplemented
+      // isWriteIn: unimplemented
+      // writeInIndex: unimplemented
+    };
+  }
+
+  return tallies;
+}
+
+function convertCandidateContest(
+  contest: ResultsReporting.CandidateContest,
+  candidateNameRecord: CandidateNameRecord
+): VxTabulation.CandidateContestResults {
+  const { overvotes, undervotes, total } = getBallotCounts(contest);
+
+  return {
+    contestId: contest['@id'],
+    contestType: 'candidate',
+    votesAllowed: contest.VotesAllowed,
+    overvotes,
+    undervotes,
+    ballots: total,
+    tallies: getCandidateTallies(contest, candidateNameRecord),
+  };
+}
+
+function convertContestsListToVxResultsRecord(
+  contestList: ReadonlyArray<
+    PartyContest | BallotMeasureContest | CandidateContest | RetentionContest
+  >,
+  candidateNameRecord: CandidateNameRecord
+): Result<VxTabulation.ElectionResults['contestResults'], Error> {
+  const vxFormattedContests: VxTabulation.ElectionResults['contestResults'] =
+    {};
+  for (const contest of contestList) {
+    if (isCandidateContest(contest)) {
+      vxFormattedContests[contest['@id']] = convertCandidateContest(
+        contest,
+        candidateNameRecord
+      );
+    } else if (isBallotMeasureContest(contest) || isRetentionContest(contest)) {
+      vxFormattedContests[contest['@id']] = convertToYesNoContest(contest);
+    } else {
+      return err(new Error(`Unsupported ERR contest type ${contest['@type']}`));
+    }
+  }
+
+  return ok(vxFormattedContests);
+}
+
+// Builds a Record of CandidateId:Name for local use. Needed because
+// ERR does not list candidate name alongside candidate ID in tallies,
+// but VX tally data structures do. This map is referenced when populating VX
+// tallies.
+function buildCandidateNameRecords(
+  electionReport: ResultsReporting.ElectionReport
+): CandidateNameRecord {
+  const records: CandidateNameRecord = {};
+  const candidates = getDeepValue(electionReport, 'Election[0].Candidate');
+
+  // Though unlikely in practice, an election may exist with no candidates
+  if (!candidates) {
+    return records;
+  }
+  for (const candidate of candidates) {
+    const textEntries = assertGetDeepValue(
+      candidate,
+      'BallotName.Text'
+    ) as ResultsReporting.LanguageString[];
+    records[candidate['@id']] = assertDefined(
+      findLanguageString(textEntries, { language: LanguageCode.ENGLISH })
+    ).Content;
+  }
+
+  return records;
+}
+/**
+ * Converts an ElectionReport to an instance of ManualElectionResults.
+ * @param electionReport
+ * @returns an instance of ManualElectionResults.
+ */
+export function getManualResultsFromErrElectionResults(
```

Comment:
nit: maybe use "convert" in the name of this function

Reply:
https://github.com/votingworks/vxsuite/pull/5344/commits/676d679685eceab7daeb635ff5782050af8a11d1